components/service/glean/src/test/java/mozilla/components/service/glean/UuidMetricTypeTest.kt

```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import java.util.UUID
+
+import mozilla.components.service.glean.storages.UuidsStorageEngine
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Test
+
+import org.junit.Before
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class UuidMetricTypeTest {
+
+    @Before
+    fun setUp() {
+        UuidsStorageEngine.clearAllStores()
+    }
+
+    @Test
+    fun `The API saves to its storage engine`() {
+        // Define a 'uuidMetric' uuid metric, which will be stored in "store1"
+        val uuidMetric = UuidMetricType(
+            applicationProperty = true,
+            disabled = false,
+            category = "telemetry",
+            name = "uuid_metric",
+            sendInPings = listOf("store1"),
+            userProperty = false
+        )
+
+        // Record two uuids of the same type, with a little delay.
+        val uuid = uuidMetric.generateAndSet()
+
+        // Check that data was properly recorded.
+        val snapshot = UuidsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+        assertEquals(1, snapshot!!.size)
+        assertEquals(true, snapshot.containsKey("telemetry.uuid_metric"))
+        assertEquals(uuid, snapshot.get("telemetry.uuid_metric"))
+
+        val uuid2 = UUID.randomUUID()
```

Comment:
Instead of using something random, let's use `UUID.fromString("a valid UUID here")`. It's not that different, but more closely resembles what the final usage scenario would probably be.